### PR TITLE
Getting Started with Ansible using GNS3

### PR DESCRIPTION
**Pull request by DilrajSK**
Hi Lindsay/Team,

I have created a guide for steps on "Getting Started with Ansible using GNS3". Kindly review it and let me know if it is adding any value.

Regards,
DILRAJ

### CODE REVIEW

File: GettingStartedwithAnsibleusingGNS3.md [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+- GNS3 VM installed on any supported Hypervisor
+- Virtual EXOS Image
+
+## Setting up GNS3 with EXOS-VM
+
+1. Download the EXOS appliance template from [GNS3 Marketplace]
+2. Download the required qcow2 file from our [GitHub Website]
+3. Import the downloaded template in GNS3 and load the qcow2 file downloaded in step 2. Visit this [link] for details
+4. Drag the installed appliance to the workspace and start it up
+5. Run few show commands on the device to make sure that its working fine
```
Comment (LindsayHill) [inline GettingStartedwithAnsibleusingGNS3.md]: I would slightly reword this. Maybe say "Run some show commands to make sure it is working. For example `show switch` and `show version`.

File: GettingStartedwithAnsibleusingGNS3.md [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+>NOTE: Default credentials to login to the device is "admin"/"". Password is blank
+
+## Setting up GNS3 with Network Automation Docker Container
+
+1. Download the Network Automation template from [GNS3 Marketplace]
+2. Import the downloaded template in GNS3 and perform the basic installation
+3. Once done, the "Network Automation" Container can be seen under the "End Devices" list
+4. Drag the docker container to the workspace and GNS3 will automatically pull the required files from its repository over the Internet
+
+> By default, the Network Automation docker container includes Python 2.7, NAPALM, pyntc, Netmiko and Ansible. So no need to install those separately.
```
Comment (LindsayHill) [inline GettingStartedwithAnsibleusingGNS3.md]: Reword "So no need to install those separately." as "You do not need to install these separately."

File: GettingStartedwithAnsibleusingGNS3.md [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+## Setting up a Test Topology
+
+Prepare the topology as show in the below image in GNS3 Workspace:
+> NOTE: **Do not** Power-on the topology at this point
+
+![Topology](http://i68.tinypic.com/5v56ch.png "Exos_Topology")
+
+### EXOS Device Setup
+
+Power-up any **one** of the EXOSVM device from the above topology and perform the below configuration.
```
Comment (LindsayHill) [inline GettingStartedwithAnsibleusingGNS3.md]: s/device/devices/

And at the end of the sentence, s/below configuration./below configuration:/

File: GettingStartedwithAnsibleusingGNS3.md [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+```
+4. Power-on the container and launch its console
+5. Once powered-on, the container should automatically get an IP address from the NAT cloud via DHCP with default subnet of 192.168.122.0/24
+>NOTE: This is the reason why we have assigned an IP address from 192.168.122.0/24 subnet to the EXOSVM.
+6. Ping the EXOSVM and make sure that it is reachable from the Automation container
+7. Run command ```ansible --version``` to check the currently installed version of Ansible
+
+## System Setup
+
+Ideally, there is no need to install ansible on Network Automation Docker Container as it already comes pre-installed with it. But for those who are using their own standalone Linux Machines can go though the below section to install the same.
```
Comment (LindsayHill) [inline GettingStartedwithAnsibleusingGNS3.md]: Should capitalize 'ansible' here. I would re-word the second sentence as "If you are using your own Linux system, you will need to go through the section below to install Ansible:"

File: GettingStartedwithAnsibleusingGNS3.md [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+### Installing Ansible
+
+1. Run the below commands to install the required files/packages:
+
+```sh
+root@NetworkAutomation-1:~# apt-get update
+root@NetworkAutomation-1:~# apt-get install git
+root@NetworkAutomation-1:~# git clone https://github.com/ansible/ansible.git --recursive
+```
+2. Do a 'ls' to make sure that ansible folder is successfully cloned. Sample command and its output is below:
```
Comment (LindsayHill) [inline GettingStartedwithAnsibleusingGNS3.md]: Change "Do a 'ls'" to "Run `ls`".

File: GettingStartedwithAnsibleusingGNS3.md [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+```sh
+root@NetworkAutomation-1:~# ls
+ansible
+```
+
+3. One can periodically pull the latest developments from Ansible's Github Repository by running the below commands:
+
+```sh
+root@NetworkAutomation-1:~# cd ansible
+root@NetworkAutomation-1:~#git pull
```
Comment (LindsayHill) [inline GettingStartedwithAnsibleusingGNS3.md]: should be space between `#` and `git pull` in that line.

File: GettingStartedwithAnsibleusingGNS3.md [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+MODULE_GUIDELINES.md  docs                      packaging         venv
+Makefile              docsite_requirements.txt  requirements.txt
+README.rst            examples                  setup.py
+```
+
+3. Activate the 'venv' virtual environment
+
+```sh
+root@NetworkAutomation-1:~#source ./ansible/venv/bin/activate
+root@NetworkAutomation-1:~#source ./ansible/hacking/env-setup
```
Comment (LindsayHill) [inline GettingStartedwithAnsibleusingGNS3.md]: Should be space between PS1 prompt (`~#`) and the commands above.

File: GettingStartedwithAnsibleusingGNS3.md [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+::1     localhost ip6-localhost ip6-loopback
+fe00::0 ip6-localnet
+ff00::0 ip6-mcastprefix
+ff02::1 ip6-allnodes
+ff02::2 ip6-allrouters
+
+192.168.122.100 S1
+```
+
+> If one has more than one switch in the network, then additional lines can be added under ```192.168.122.100 S1```
```
Comment (LindsayHill) [inline GettingStartedwithAnsibleusingGNS3.md]: Reword as "If you have more than one virtual switch in your network, then additional lines..."

File: GettingStartedwithAnsibleusingGNS3.md [hunk 9]
```diff
@@ -0,0 +1,283 @@
...
+---
+ansible_network_os: exos
+ansible_connection: network_cli
+ansible_user: xtrm_user
+ansible_ssh_pass: xtrm_pass
+```
+
+## Testing the Setup
+
+1. To confirm your credentials, connect to a network device(EXOS-VM) manually using command ```ssh xtrm_user@S1``` and run few show comamnds to verify the same
```
Comment (LindsayHill) [inline GettingStartedwithAnsibleusingGNS3.md]: s/comamnds/commands/

**Review by LindsayHill — APPROVED** [review]
Very good work. Feedback is all just very minor editing things.

**Comment by LindsayHill** [comment]
This is really good work, and much appreciated. My review comments are all just minor editing things, no big deal. 

One other thing we should add here is an update to the top-level README, linking to this document. Can you include that in this PR?

File: GettingStartedwithAnsibleusingGNS3.md (hunk 1 above)
Comment (DilrajSK) [reply GettingStartedwithAnsibleusingGNS3.md]: done

File: GettingStartedwithAnsibleusingGNS3.md (hunk 2 above)
Comment (DilrajSK) [reply GettingStartedwithAnsibleusingGNS3.md]: done

File: GettingStartedwithAnsibleusingGNS3.md (hunk 3 above)
Comment (DilrajSK) [reply GettingStartedwithAnsibleusingGNS3.md]: done

File: GettingStartedwithAnsibleusingGNS3.md (hunk 4 above)
Comment (DilrajSK) [reply GettingStartedwithAnsibleusingGNS3.md]: done

File: GettingStartedwithAnsibleusingGNS3.md (hunk 5 above)
Comment (DilrajSK) [reply GettingStartedwithAnsibleusingGNS3.md]: done

File: GettingStartedwithAnsibleusingGNS3.md (hunk 6 above)
Comment (DilrajSK) [reply GettingStartedwithAnsibleusingGNS3.md]: done

File: GettingStartedwithAnsibleusingGNS3.md (hunk 7 above)
Comment (DilrajSK) [reply GettingStartedwithAnsibleusingGNS3.md]: done

File: GettingStartedwithAnsibleusingGNS3.md (hunk 8 above)
Comment (DilrajSK) [reply GettingStartedwithAnsibleusingGNS3.md]: done

File: GettingStartedwithAnsibleusingGNS3.md (hunk 9 above)
Comment (DilrajSK) [reply GettingStartedwithAnsibleusingGNS3.md]: done

**Comment by DilrajSK** [comment]
Glad that you liked it and thank you for your email to my manager. 

I have included a line at the bottom of EXOS section linking to this document. But the link which I have included is a link from the forked repository. Do you think that it's going to work that way? The link looks like below:

https://github.com/DilrajSK/ansible-extreme/blob/master/GettingStartedwithAnsibleusingGNS3.md

**Comment by LindsayHill** [comment]
I think the trick here will be to have a relative URL - e.g. ./GettingStartedwithAnsibleusingGNS3.md

That way it will work no matter what repo it is in

**Comment by DilrajSK** [comment]
It's done.

**Comment by LindsayHill** [comment]
Thanks, merged

**Comment by DilrajSK** [comment]
Thanks. Appreciate it

**Pull request merged**